service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractStatusServiceUnitTest.java

```diff
@@ -0,0 +1,142 @@
+package uk.nhs.adaptors.gp2gp.ehr;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.mongodb.core.FindAndModifyOptions;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.Update;
+import uk.nhs.adaptors.gp2gp.common.service.TimestampService;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus;
+import uk.nhs.adaptors.gp2gp.mhs.exception.UnrecognisedInteractionIdException;
+import java.time.Duration;
+
+import java.time.Instant;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class EhrExtractStatusServiceUnitTest {
```

Comment:
This is interesting, I always assumed that `*Test.java` were unit tests - I've not noticed until now that some are labelled `*UnitTest.java`.

Not precious about this one, but leaning towards `EhrExtractStatusServiceTest` for consistency!

Reply:
It was done predominantly to avoid the clash with the same named class in integration test. If they had the same names the tests would behave weirdly, hence I added that "Unit" thing to differentiate it.